tests/rptest/services/redpanda.py

```diff
@@ -709,6 +715,10 @@ def _load_s3_context(self, logger, test_context):
         else:
             logger.info('No AWS credentials supplied, assuming minio defaults')
 
+        # Override endpoint if it is set by cloud_storage_api_endpoint
+        if self.has_cloud_storage_api_endpoint_override:
+            self.endpoint_url = f'http://{self.cloud_storage_api_endpoint}:{self.cloud_storage_api_endpoint_port}'
```

Comment:
This code is very confusing. The fact that we set `self.endpoint_url` above then override it again is pretty bad.

I acknowledge that it was quite bad before you made changes but **if you have time** to think of a way that would avoid this convoluted flow and duplication in logic I'd like to ask you to take a second look at it.